bancor_research/bancor_simulator/v3/spec/actions.py

```diff
@@ -376,23 +376,34 @@ def process_withdrawal(
     return state
 
 
-def begin_withdrawal_cooldown(state, withdraw_value, tkn_name, user_name):
+def begin_withdrawal_cooldown(
+    state, withdraw_value, tkn_name, user_name, by_ptkn_amt: bool = False
```

Comment:
suggest renaming to `is_unit_tkn` (or similar). also if you're adding type hinting here (and we should everywhere imo), why not add to the rest of the variables? should be as follows:

`state: State, withdraw_value: Any, tkn_name: str, user_name: str, is_unit_tkn: bool = False`

Reply:
TBH, I'm reconsidering the whole boolean-flag thing.
Maybe better off splitting the function into:
- `begin_withdrawal_cooldown_by_reserve_token`
- `begin_withdrawal_cooldown_by_pool_token`

?